independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/AsyncInvokerImpl.java

```diff
@@ -45,12 +47,30 @@ public AsyncInvokerImpl(ClientImpl restClient, HttpClient httpClient, URI uri, R
         this.httpClient = httpClient;
         this.uri = uri;
         this.requestSpec = new RequestSpec(requestSpec);
+        addUserInfoIfNecessary(this.uri, this.requestSpec);
         this.configuration = configuration;
         this.properties = new HashMap<>(properties);
         this.handlerChain = handlerChain;
         this.requestContext = requestContext;
     }
 
+    private void addUserInfoIfNecessary(URI uri, RequestSpec requestSpec) {
+        String userInfo = uri.getUserInfo();
+        if (userInfo == null) {
+            return;
+        }
+        String[] parts = userInfo.split(":");
+        if (parts.length != 2) {
+            return;
+        }
+        ClientRequestHeaders specHeaders = requestSpec.headers;
```

Comment:
I have tested manually using POSTman and this is equivalent to put the user info (`USER:PASSWORD`) in the URL :white_check_mark: